pkg/cloudprovider/provider/openstack/provider.go

```diff
@@ -565,53 +579,50 @@ func (p *provider) Create(machine *v1alpha1.Machine, data *cloudprovidertypes.Pr
 		}
 	}
 
-	if err := p.serverReadinessWaiter(computeClient, server.ID, c.InstanceReadyCheckPeriod, c.InstanceReadyCheckTimeout); err != nil {
-		defer deleteInstanceDueToFatalLogged(computeClient, server.ID)
-		return nil, fmt.Errorf("instance %s became not active: %v", server.ID, err)
-	}
-
-	// Find a free FloatingIP or allocate a new one
 	if c.FloatingIPPool != "" {
```

Comment:
poll and wait ONLY IN CASE if it's really needed (i.e. FloatingIPPool is set and thus floating IP is required).